### PR TITLE
Hot fix auto loading of funnels

### PR DESCRIPTION
**Pull request by alexkim205**
## Changes

Missed an edge case in #5597 where funnels weren't loading on parameter changes (bin count changes, conversion window change, etc). Ideally should refactor this so that these parameters live in filters, but this should temporarily fix it for now.

- [x] Bring histogram steps, bin counts, and conversion windows into filters
- [x] Cleanup filter construction in `funnelLogic.ts`
- [x] Remove problematic if check
- [x] Better fix for #5576

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by macobo** [comment]
Perhaps worth reverting https://github.com/PostHog/posthog/pull/5576 instead and re-implementing it?

Note my PR only fixed one edge case - it's pretty weird we have this behavior for funnels specifically.

**Comment by macobo** [comment]
Perhaps worth reverting https://github.com/PostHog/posthog/pull/5576 instead and re-implementing it?

Note my PR only fixed one edge case - it's pretty weird we have this behavior for funnels specifically and might need proper consideration.

cc @mariusandra

**Comment by mariusandra** [comment]
I think this is solved in https://github.com/PostHog/posthog/pull/5612

If the insight is initially through a call to the API endpoint `/api/insight/534234` (e.g. happens when `#fromItem=534234&fromDashboard=2` is in the url), the `lastAppliedFilters` get set, without them actually having been applied. I refactored the value to use a selector instead of needing to be explicitly set.

**Comment by alexkim205** [comment]
@mariusandra, just tested out #5612 and I think your changes would've fixed this as well if conversion window and bin counts were persisted in `filters` but it looks like they all exist as their own selectors/reducers. This strays from our data structures in other `{insight}Logic`'s, and additionally prevents these parameters from being persisted in the url.

@macobo I'll remove that `if` check for now and look for a better way to refactor in this PR.

P.S. @mariusandra ~~is there a best practices pattern to ignore specific parameters in filters without introducing a hacky if check in `loadResults`?~~

Actually I think the best place to put this check is here where we do something similar already with `shouldRefresh` - https://github.com/PostHog/posthog/blob/fix/load-funnels/frontend/src/scenes/funnels/funnelLogic.ts#L622-L622

**Comment by alexkim205** [comment]
Ok! Updated the description with a checklist of things I did in this PR. These changes should address:

1. don't refresh on layout #5570
2. refresh failing #5597
3. persists bin count, histogram steps, and conversion window to URL and filters

**Comment by alexkim205** [comment]
cc @mariusandra for review

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 1]
```diff
@@ -624,8 +608,13 @@ export const funnelLogic = kea<funnelLogicType>({
...
             const shouldRefresh =
                 values.filters?.events?.length === 2 && values.lastAppliedFilters?.events?.length === 1
+            // If layout is the only thing that changes
+            const shouldNotRefresh = equal(
+                Object.assign({}, values.filters, { layout: undefined }),
+                Object.assign({}, values.lastAppliedFilters, { layout: undefined })
+            )
 
-            if (refresh || shouldRefresh || clickhouseFeaturesEnabled) {
+            if ((refresh || shouldRefresh || clickhouseFeaturesEnabled) && !shouldNotRefresh) {
```
Comment (mariusandra) [inline frontend/src/scenes/funnels/funnelLogic.ts:617]: This line feels a bit messy. `shouldRefresh && !shouldNotRefresh` :D.

Perhaps rename the double negation var to `onlyLayoutChanged`?

**Review by mariusandra — APPROVED** [review]
I think this makes sense, just a nit below.

**Pull request merged**